synapse/handlers/room_member.py

```diff
@@ -888,7 +888,8 @@ def _is_host_in_room(self, current_state_ids):
             if not event:
                 continue
 
-            if event.membership == Membership.JOIN:
+            if (event.membership == Membership.JOIN
+                    or event.membership == Membership.INVITE):
```

Comment:
this doesn't look right. servers with no users in the `join` state won't receive updates in the room, so need to do the full `make_join` dance to get an up-to-date room DAG.

Reply:
I understand the problem. I'm saying this will break more than it fixes.

Reply:
For example, what can break?

Reply:
you'll end up with out-of-date room state.

Reply:
if 3 people are invited to the room, two accepted the invitation, and the third accepted later, after talking about something and leaving the room, the third should chat chat after accepting the invitation; give to the list of available

Reply:
I understand the problem. I'm saying this will break more than it fixes.